CMakeLists.txt

```diff
@@ -267,6 +267,13 @@ if(HARDENING)
     try_append_cxx_flags("-fcf-protection=full" TARGET hardening_interface)
 
     if(MINGW)
+      # Older versions of gcc required libssp to be linked, but this was fixed
+      # with gcc-11. This can be removed when those older versions are no longer
+      # supported.
+      if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 11.0)
```

Comment:
My first instinct was to complain about having this hard-coded and instead request that we check it instead, but it's a specific lib tied to a compiler implementation, so in this case I think the self-documenting test is fine.

Though... From a quick search I can't actually find where this was fixed in gcc11. Could you please link the bug/discussion/commit that shows this was a purposeful fix and that we don't need to worry about it regressing in the future? It'd be helpful to have that documented here so that it's straightforward to remove this chunk once we require gcc >= 12.

Reply:
I could only refer to this https://github.com/bitcoin/bitcoin/pull/28461#issuecomment-1717216829.

Reply:
Ah, so it's mingw-w64 v11, not gcc. Could you update the comment and link that mailing list post?

Reply:
It seems, it is not correct to correlate it to `CMAKE_CXX_COMPILER_VERSION`, right?

Reply:
Ah yes, I suppose that's true. I suppose this should be a compile test after all.

An example for distinction: consider that we could (in the future) be building with clang + mingw-w64.

Reply:
> I suppose this should be a compile test after all.

Done.